requirements.txt

```diff
@@ -0,0 +1,5 @@
+dash>=1.20.0
```

Comment:
I agree that pinned dependencies give you 100% reproducibility, but  they also need to be maintained and frequently updated in order to keep up with security and other fixes. Unless this can be automated, I actually prefer to specify just the minimal required version (especially in the context of an example application). We though should list (as part of dev documentation) actual versions of dependencies with which the code was tested. Do you have any experience in automating update pinned dependencies?

Reply:
Updating dependencies could be automated with dependabot GH Action: https://docs.github.com/en/code-security/supply-chain-security/about-dependabot-version-updates#supported-repositories-and-ecosystems It supports Python.
We have it configured for Synthetics provider and it works well so far: https://github.com/kentik/terraform-provider-kentik-synthetics/blob/master/.github/dependabot.yml 

I think that we should freeze deps in requirements.txt file and try Dependabot. It would be helpful `requirements-to-freeze.txt` to keep track of direct dependencies (see: https://kenreitz.org/essays/2016/02/25/a-better-pip-workflow)

Additionaly we could use https://pipenv.pypa.io/en/latest/ which seems like the best tools for managing dependencies in Python.